python_modules/libraries/dagster-mariadb/dagster_mariadb/partitioned_io_manager.py

```diff
@@ -0,0 +1,282 @@
+            if self.partition_column not in columns:
+                columns[self.partition_column] = "VARCHAR(255)"
+            
+            # Build partition config based on Dagster partition definition
+            partition_config = self._build_partition_config(context)
+            
+            create_partitioned_table(
+                table_name=table_name,
+                schema_name=self.schema_name,
+                columns=columns,
+                partition_config=partition_config,
+                connection=connection,
+                storage_engine=self.storage_engine,
+            )
+            context.log.info(f"Created partitioned table {self._get_full_table_name(context)}")
+
+    def _build_partition_config(self, context: OutputContext) -> Dict[str, Any]:
+        """Build partition configuration from Dagster partition definition."""
+        if not hasattr(context, 'asset_partitions_def') or context.asset_partitions_def is None:
+            # Default to hash partitioning if no partition definition
+            return PartitionStrategy.hash_partitions(
+                partition_column=self.partition_column,
+                num_partitions=10
+            )
+        
+        from dagster import DailyPartitionsDefinition, MonthlyPartitionsDefinition, StaticPartitionsDefinition
+        
+        partitions_def = context.asset_partitions_def
+        
+        if isinstance(partitions_def, DailyPartitionsDefinition):
+            return PartitionStrategy.daily_to_range(
+                partition_column=self.partition_column,
+                start_date=partitions_def.start,
+                end_date=partitions_def.end if hasattr(partitions_def, 'end') and partitions_def.end else "2099-12-31",
+            )
+        elif isinstance(partitions_def, MonthlyPartitionsDefinition):
+            return PartitionStrategy.monthly_to_range(
+                partition_column=self.partition_column,
+                start_date=partitions_def.start,
+                end_date=partitions_def.end if hasattr(partitions_def, 'end') and partitions_def.end else "2099-12-31",
+            )
+        elif isinstance(partitions_def, StaticPartitionsDefinition):
+            # Use LIST partitioning for static partitions
+            partition_keys = partitions_def.get_partition_keys()
+            partition_values = {key: [key] for key in partition_keys}
+            return PartitionStrategy.list_partitions(
+                partition_column=self.partition_column,
+                partition_values=partition_values
+            )
+        else:
+            # Default to hash partitioning
+            return PartitionStrategy.hash_partitions(
+                partition_column=self.partition_column,
+                num_partitions=10
+            )
+
+    def _infer_sql_type(self, dtype) -> str:
+        """Infer SQL type from pandas dtype."""
+        dtype_str = str(dtype)
+        if 'int' in dtype_str:
```

Comment:
`int` is "in" `print`, so that's not correct. Types can also be user-defined. Parse it properly. I suppose it follows the pattern `TYPE(\(<value>\))?` (not really a regexp). startswith+follows by non-word would do (i hope O:-) ).